### ISSUE TITLE
Backlog CRM Analytics

### ISSUE THREAD

**Issue by yosra-wanen** [yosra-wanen/lynara-ai-marketing-campaigns#14]
# Backlog Business Analytics - Plateforme Multicanale IA

## 1) Axes Business Analytics a prioriser

- Funnel Social -> CRM -> Conversion: source, campagne, canal, statut lead, conversion, delai de conversion.
- Qualite des leads: completude, enrichissement, deduplication, verification email/telephone, scoring.
- Performance contenu/canal: engagement (reach, likes, comments, saves, shares), CTR, reponse DM/comment, cout par lead.
- Pilotage operationnel: SLA agents (latence, retries, erreurs), quotas API, cout IA par canal/campagne.
- ROI marketing: budget campagne vs valeur convertie (`estimated_value`, conversion reelle, CPL/CAC).

### A1. Modele analytique unifie
- Creer un modele evenementiel: `analytics.events` (lead_created, message_sent, post_published, dm_received, lead_converted).
- Sortie attendue: vue `analytics.v_funnel_social_business`.

### A2. Tracking et attribution
- Ajouter attribution `source_channel`, `campaign_id`, `utm_*`, `touchpoint_id` sur leads/interactions.
- Sortie attendue: first-touch / last-touch par lead.

### A3. Qualite et conformite
- Regles DQ: email format, telephone normalise E.164, doublons (email+phone+company).
- Sortie attendue: score qualite journalier par company.

### B1. Lead scoring predictif
- Score 0-100 combinant profil, activite, engagement social, historique CRM.
- Sortie attendue: `crm.lead_scores` + raisons explicatives.

### B2. Segmentation dynamique
- Segments automatiques (RFM-like + comportemental + geo + intention).
- Sortie attendue: rafraichissement quotidien des segments.

### B3. Probabilite de conversion
- Modele de classification + horizon temps (7/30 jours).
- Sortie attendue: `conversion_probability`, `expected_revenue`.

### C1. KPI campagne multicanal
- KPIs: delivery/open/click/reply (email/WA), engagement social, CPL, conversion rate.
- Sortie attendue: dashboard "Campaign Performance".

### C2. A/B testing analytics
- Comparer caption/tone/langue/horaire/format.
- Sortie attendue: recommandation "next best variant".

### C3. Coherence contenu
- Mesurer qualite texte (lisibilite, ton, langue) et pertinence segment.
- Sortie attendue: score content-quality par message.

### D1. Listener + intent detection
- Classer commentaires/DM: question prix, disponibilite, visite, objection.
- Sortie attendue: `intent`, `urgency`, `lead_created_flag`.

### D2. Conversion social
- Creer lead depuis interaction sociale + lien avec post/source.
- Sortie attendue: taux de conversion par plateforme (IG/FB/TikTok).

### D3. Competitive benchmark
- Indicateurs frequence/format/engagement des concurrents.
- Sortie attendue: recommandations hebdomadaires actionnables.

### E1. Executive dashboard
- Vue direction: pipeline, revenue, ROI, CPL/CAC, top channels.

### E2. Ops dashboard
- Vue equipe: queue agents, erreurs, SLA, posts a relancer.

### E3. BI export
- Vues SQL stables pour Power BI.

## 3) Scripts SQL de migration proposes

```sql
-- 001_analytics_events.sql
create schema if not exists analytics;

create table if not exists analytics.events (
  event_id uuid primary key default gen_random_uuid(),
  company_id uuid not null,
  event_type text not null,
  event_ts timestamptz not null default now(),
  lead_id uuid null,
  campaign_id uuid null,
  channel text null,               -- instagram, facebook, tiktok, email, whatsapp
  source_platform text null,       -- ig, fb, tt
  source_id text null,             -- external id post/message
  session_id text null,
  user_id uuid null,
  properties jsonb not null default '{}'::jsonb
);

create index if not exists idx_events_company_ts on analytics.events(company_id, event_ts desc);
create index if not exists idx_events_type_ts on analytics.events(event_type, event_ts desc);
create index if not exists idx_events_lead on analytics.events(lead_id);
```

```sql
-- 002_crm_lead_attribution.sql
alter table crm.leads
  add column if not exists acquisition_channel text,
  add column if not exists source_platform text,
  add column if not exists source_campaign_id uuid,
  add column if not exists first_touch_at timestamptz,
  add column if not exists last_touch_at timestamptz,
  add column if not exists conversion_value numeric(18,2) default 0;

create index if not exists idx_leads_acq_channel on crm.leads(company_id, acquisition_channel);
create index if not exists idx_leads_source_campaign on crm.leads(company_id, source_campaign_id);
```

```sql
-- 003_social_interactions.sql
create schema if not exists conversations;

create table if not exists conversations.social_interactions (
  interaction_id uuid primary key default gen_random_uuid(),
  company_id uuid not null,
  platform text not null,           -- instagram/facebook/tiktok
  interaction_type text not null,   -- comment, dm, reaction
  external_post_id text null,
  external_message_id text null,
  account_id uuid null,
  lead_id uuid null,
  content text null,
  intent text null,
  sentiment text null,
  urgency_score integer null check (urgency_score between 0 and 100),
  created_at timestamptz not null default now()
);

create index if not exists idx_social_company_created on conversations.social_interactions(company_id, created_at desc);
create index if not exists idx_social_platform_type on conversations.social_interactions(company_id, platform, interaction_type);
```

```sql
-- 004_lead_scores_history.sql
create table if not exists crm.lead_scores (
  lead_score_id uuid primary key default gen_random_uuid(),
  lead_id uuid not null references crm.leads(lead_id) on delete cascade,
  company_id uuid not null,
  score integer not null check (score between 0 and 100),
  model_version text not null,
  score_reason jsonb not null default '{}'::jsonb,
  predicted_conversion_probability numeric(5,2) null,
  scored_at timestamptz not null default now()
);

create index if not exists idx_lead_scores_company_time on crm.lead_scores(company_id, scored_at desc);
create index if not exists idx_lead_scores_lead_time on crm.lead_scores(lead_id, scored_at desc);
```

```sql
-- 005_analytics_views.sql
create or replace view analytics.v_funnel_social_business as
select
  l.company_id,
  date_trunc('day', l.created_at) as day,
  coalesce(l.source_platform, 'unknown') as source_platform,
  count(*) as leads_created,
  count(*) filter (where l.status in ('qualified','proposal','negotiation','converted')) as leads_qualified_plus,
  count(*) filter (where l.is_converted = true) as leads_converted,
  sum(coalesce(l.estimated_value,0)) as estimated_pipeline
from crm.leads l
group by 1,2,3;
```

## Front (Next.js + TypeScript)

- Page `dashboard/executive`:
  - KPI cards: Leads, Qualified, Converted, Conversion Rate, CPL, ROI.
  - Graphiques: funnel, evolution hebdo, top canaux.
- Page `dashboard/campaigns`:
  - Table campagnes + filtres (canal, langue, segment, periode).
  - A/B results panel (variant winner + uplift).
- Page `dashboard/social`:
  - Engagement par plateforme + intents detectees + social->lead conversion.
- Page `dashboard/leads-intelligence`:
  - Distribution scores, segments dynamiques, explication score lead.
- Page `dashboard/ops`:
  - Erreurs pipeline, retries, latence jobs, quotas API.

## Back (FastAPI)

- `GET /analytics/funnel?from&to&company_id`
- `GET /analytics/campaign-performance`
- `GET /analytics/social-conversion`
- `GET /analytics/lead-scoring-distribution`
- `GET /analytics/roi-report`
- `POST /ml/score-leads` (batch)
- `POST /ml/recompute-segments`
- `GET /analytics/export/powerbi/*` (vues materialisees ou SQL views stables)

## 5) Algorithmes ML recommandes (par partie)

- Lead scoring: XGBoost / LightGBM (tabulaire), fallback Logistic Regression explicable.
- Probabilite de conversion: classification binaire (CatBoost/XGBoost) + calibration (Platt/Isotonic).
- Delai de conversion: modele de survie (Cox PH) ou regression (XGBoost Regressor).
- Segmentation leads: KMeans (baseline), HDBSCAN (clusters non spheriques), + PCA/UMAP pour visualisation.
- Intent commentaires/DM: CamemBERT ou mBERT fine-tuned multiclass.
- Sentiment social: modele transformer multilingue FR/AR/EN.
- Recommandation horaire/format: bandits contextuels (LinUCB, Thompson Sampling).
- Anomalies KPI: Isolation Forest ou Prophet + detection residus.
- Attribution marketing: Markov chain attribution + baseline first/last touch.

### T1 - Funnel end-to-end
- Input: post IG + DM + creation lead + conversion.
- Attendu: tracabilite complete dans `analytics.events` et dashboard mis a jour.

### T2 - Attribution multicanale
- Input: meme lead touche par IG puis WhatsApp.
- Attendu: first-touch IG et last-touch WA corrects.

### T3 - Lead scoring
- Input: lead enrichi + interactions fortes.
- Attendu: score augmente et `score_reason` coherent.

### T4 - Segmentation
- Input: batch de leads multi-profils.
- Attendu: segments stables et business-relevant.

### T5 - A/B testing
- Input: deux captions, meme audience.
- Attendu: uplift exploitable et winner recommande.

### T6 - Data quality
- Input: doublons email/tel, formats invalides.
- Attendu: blocage/flag + rapport DQ.

### T7 - Resilience pipeline
- Input: indisponibilite API Instagram.
- Attendu: retries + DLQ + alerting + aucune duplication.

### T8 - Performance
- Input: 100k events/jour.
- Attendu: requetes dashboard < 2s avec vues/index adaptes.

### T9 - Securite/RLS
- Input: requetes multi-tenant.
- Attendu: isolation stricte par `company_id`.

### T10 - Drift ML
- Input: changement distribution features.
- Attendu: alerte drift + plan de recalibration.

## 7) Suggestions complementaires

- Mettre en place un dictionnaire de metriques versionne (definition KPI, formule, owner).
- Ajouter une couche "feature store" simple pour reutiliser les variables ML entre training/inference.
- Ajouter des tests de non-regression data (dbt tests ou SQL assertions) en CI.
- Prevoir un monitoring cout IA par modele/canal pour arbitrage budgetaire.
